src/dashboards/videos/details.jsonnet

```
@@ -1,17 +1,121 @@
 local grafana = import 'grafonnet/grafana.libsonnet';
 local dashboard = grafana.dashboard;
+local elasticsearch = grafana.elasticsearch;
+local graphPanel = grafana.graphPanel;
 local template = grafana.template;
 
+
+// Constants
+local lrs = 'lrs';
+local course_field = 'object.definition.extensions.http://adlnet.gov/expapi/activities/course.keyword';
+local school_field = 'object.definition.extensions.https://w3id.org/xapi/acrossx/extensions/school.keyword';
+local session_field = 'object.definition.extensions.http://adlnet.gov/expapi/activities/module.keyword';
+local video_id_field = 'object.id.keyword';
+
+
+// Utils
+local double_escape_string(x) = std.strReplace(std.strReplace(x, ':', '\\\\:'), '/', '\\\\/');
+
+
+// Dashboard
 dashboard.new(
   'Details',
   tags=['xAPI', 'video', 'teacher'],
   editable=false
 )
 .addTemplate(
-  template.datasource(
-    'PROMETHEUS_DS',
-    'prometheus',
-    'Prometheus',
-    hide='label',
+  template.new(
+    name='SCHOOL',
+    current='all',
+    label='School',
+    datasource=lrs,
+    query='{"find": "terms", "field": "%(school)s"}' % {
+      school: school_field,
+    },
+    refresh='time'
+  )
+)
+.addTemplate(
+  template.new(
+    name='COURSE',
+    current='all',
+    label='Course',
+    datasource=lrs,
+    query='{"find": "terms", "field": "%(course)s", "query": "%(school)s:$SCHOOL"}' % {
+      course: course_field,
+      school: double_escape_string(school_field),
+    },
+    refresh='time'
   )
 )
+.addTemplate(
+  template.new(
+    name='SESSION',
+    current='all',
+    label='Session',
+    datasource=lrs,
+    query='{"find": "terms", "field": "%(session)s", "query": "%(course)s:$COURSE"}' % {
+      session: session_field,
+      course: double_escape_string(course_field),
+    },
+    refresh='time'
+  )
+)
+.addTemplate(
+  template.new(
+    name='VIDEO',
+    current='all',
+    label='Video',
+    datasource=lrs,
+    query='{"find": "terms", "field": "%(video_id)s", "query": "%(course)s:$COURSE AND %(session)s:$SESSION"}' % {
+      video_id: video_id_field,
+      course: double_escape_string(course_field),
+      session: double_escape_string(session_field),
+    },
+    refresh='time'
+  )
+)
+.addPanel(
+  graphPanel.new(
+    title='Verbs',
```

This is optional but it would be nice to have a `description` field for this panel. Something like:
```suggestion
    title='Verbs',
    description=|||
        Here we show the distribution of video events by their type and by hour.
    |||,
```

---

Totally. I think we could add this description once we have a better idea of its content (as there is only one graph currently). WDYT?

---

Sounds good) Apropos, this description will be attached to the graph (it appears when we hover over the top left corner of the graph)